frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { PlusCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import { Button, ButtonVariant } from '@patternfly/react-core';
+import ProbeForm from './ProbeForm';
+import { HealthCheckProbeDataType } from './health-checks-types';
+import { getHealthChecksProbe } from './health-check-probe-utils';
+import { getValidationSchema } from './health-check-probe-validation-utils';
+import './HealthCheckProbe.scss';
+
+interface HealthCheckProbeProps {
+  probe: string;
+  initialValues: HealthCheckProbeDataType;
+}
+
+const HealthCheckProbe: React.FC<HealthCheckProbeProps> = ({ probe, initialValues }) => {
+  const [addProbe, setAddProbe] = React.useState(false);
+  const [selectedProbe, setSelectedProbe] = React.useState<React.ReactNode>('');
+  const [probeData, setProbeData] = React.useState<HealthCheckProbeDataType>();
+
+  const onEditProbe = () => {
+    setAddProbe(true);
+  };
+
+  const handleDeleteProbe = () => {
+    setAddProbe(false);
+    setSelectedProbe('');
+    setProbeData(initialValues);
+  };
+
+  const handleReset = (values, actions) => {
+    actions.resetForm({ values: initialValues, status: {} });
+    setAddProbe(false);
+  };
+
+  const handleSubmit = (values, actions) => {
+    actions.setSubmitting(true);
+    setAddProbe(false);
+    setProbeData(values);
+    setSelectedProbe(
+      <>
+        {`${getHealthChecksProbe(probe).formTitle} Added`}
+        <Button
+          className="pf-m-plain--align-left"
+          variant={ButtonVariant.plain}
+          onClick={handleDeleteProbe}
+        >
+          <MinusCircleIcon />
+        </Button>
+      </>,
+    );
+  };
+
+  const handleAddProbe = () => {
+    setAddProbe(true);
+  };
+
+  const renderProbe = () => {
+    if (addProbe) {
+      return (
+        <Formik
```

Comment:
@debsmita1 I dont' think we need this ```Formik``` here for probe form. cc: @rohitkrai03

Reply:
How would you integrate the forms inside the actual form or advanced options of git import if you have created separate formik instances of each section?

Reply:
`Formik` inside of `Formik` is not a usual pattern that we want to follow unless its absolutely necessary. In cases where you want the form to be actually submitted and create some resources or something like we do with create secrets modal, then its okay. But not in this case.